coordinator/src/main/java/com/tencent/rss/coordinator/ApplicationManager.java

```diff
@@ -161,21 +163,23 @@ protected synchronized void incRemoteStorageCounter(String remoteStoragePath) {
 
   @VisibleForTesting
   protected synchronized void decRemoteStorageCounter(String storagePath) {
-    AtomicInteger atomic = remoteStoragePathCounter.get(storagePath);
-    if (atomic != null) {
-      int count = atomic.decrementAndGet();
-      if (count < 0) {
-        LOG.warn("Unexpected counter for remote storage: %s, which is %i, reset to 0",
-            storagePath, count);
-        atomic.set(0);
+    if (!StringUtils.isEmpty(storagePath)) {
+      AtomicInteger atomic = remoteStoragePathCounter.get(storagePath);
+      if (atomic != null) {
+        int count = atomic.decrementAndGet();
+        if (count < 0) {
+          LOG.warn("Unexpected counter for remote storage: %s, which is %i, reset to 0",
+              storagePath, count);
+          atomic.set(0);
+        }
+      } else {
+        LOG.warn("Can't find counter for remote storage: {}", storagePath);
+        remoteStoragePathCounter.putIfAbsent(storagePath, new AtomicInteger(0));
+      }
+      if (remoteStoragePathCounter.get(storagePath).get() == 0
```

Comment:
Is it possible that remoteStoragePathCounter.get  return null?

Reply:
if null, there will be `remoteStoragePathCounter.putIfAbsent()` to add new object before this statement.